### ISSUE TITLE
ca65: .INCLUDE is also broken within .REPEAT

### ISSUE THREAD

**Issue by kugelfuhr** [cc65/cc65#2159]
It is known that .INCLUDE is broken when used within a macro. The behavior is that all include statements within the macro are not executed when the macro is expanded, but when macro expansion is over. So actually they are executed out of order.

I found that this is also true for .REPEAT. This code:

```
$ cat test.inc
        lda     #sym
$ cat test.s
.repeat  10, i
        sym     .set    i
        .include        "test.inc"
.endrep
```

does not generate 10 `lda` instructions with an increasing immediate operand, but 10 `lda` instructions that load the immediate value 9.

The behavior of .INCLUDE within macros was changed in #1477 in an attempt to "fix" it. It was found that this change breaks existing code in #1503 and #1655 and then reverted. If it will ever get fixed, this has to be done for .REPEAT also.

**Comment by bbbradsmith** [comment]
Related to #2155

**Comment by mrdudz** [comment]
This never worked, right?

**Comment by kugelfuhr** [comment]
> This never worked, right?

Since it seems to be related to .INCLUDE handling which has always been broken, I would say yes. But I don't know for sure.

**Comment by mrdudz** [comment]
closed via #2498
